From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Luke=20Sern=C3=A9?= <luke.serne@solconmail.nl>
Date: Thu, 25 Dec 2025 19:06:05 +0100
Subject: [PATCH] 8827: Decompiler: Don't restrict long integer literals

Decompiler: Don't restrict long integer literals Previously,
 `CastStrategy::markExplicitLongSize` only applied to the first input of shift
 operations. Using "L" suffixes is more widely useful, and sometimes even
 required for correct decompilation output: "-0x80000000" and "-0x80000000L"
 compile to very different constants.
---
 Ghidra/Features/Decompiler/src/decompile/cpp/cast.cc | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/cast.cc b/Ghidra/Features/Decompiler/src/decompile/cpp/cast.cc
index fbc7a0bda0..754162ac72 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/cast.cc
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/cast.cc
@@ -79,8 +79,6 @@ bool CastStrategy::markExplicitUnsigned(PcodeOp *op,int4 slot) const
 bool CastStrategy::markExplicitLongSize(PcodeOp *op,int4 slot) const
 
 {
-  if (!op->getOpcode()->isShiftOp()) return false;
-  if (slot != 0) return false;
   Varnode *vn = op->getIn(slot);
   if (!vn->isConstant()) return false;
   if (vn->getSize() <= promoteSize) return false;
-- 
2.45.1

